numpy/testing/_private/utils.py

```diff
@@ -704,7 +706,7 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
         # compared usefully, and for which .all() yields masked.
         x_id = func(x)
         y_id = func(y)
-        if (x_id == y_id).all() != True:
+        if npany(x_id != y_id):
```

Comment:
Would it be overkill to mock up an `ndarray` subclass for which this succeeds but the previous version fails for a test of the test?